### ISSUE TITLE
Mentor details

### ISSUE THREAD

**Issue by edmondpr** [MentorsWithoutBorders/mwb-partners-admin#63]
**Design:** https://www.figma.com/file/DgrKZX9mS13Zgf7k5pZ1WT/Partner-web-app?type=design&node-id=727-148&t=LPfyjEb31x9adSCp-4

When clicking on the details icon for a mentor in the mentors table, an overlay with the mentor's full name as title and  a list of courses for that mentor will be shown, with the following details for each course:
- duration of the course, start date - end date in parenthesis, and the number of hours for the lessons completed so far
- _Students_ subtitle
- the list of students enrolled in the course, with the student's full name, organization name
- in a parallel column, the buttons for opening 1 or 2 video testimonials for each student will be visible (if the testimonials exist)
- each testimonial button will have a text with the date of the testimonial inside (in the MMM DD, YYYY format) and the play icon on the right-hand side of the date
- when pressing a testimonial button, the YouTube video with that testimonial will appear in the same window, replacing the details. A back icon on the top left side will allow the user to return from the video to the details.

**Comment by ugudlado** [comment]
@edmondpr 
> the project associated with that course (if any)

There is no direct relation between project and course. We can get project via mentors but Mentor - Project is many-many relation.  Is there any other way or table to check if there is any project associated with a course?

>the number of hours corresponding to the lessons completed so far (or within the date range specified for the mentors table)
I tried to check in the code to get this but couldn't find any reference of hours related to completed lessons.
Please help in providing direction to get this number. Which tables should be referred here?

**Comment by edmondpr** [comment]
> > the project associated with that course (if any)
> 
> There is no direct relation between project and course. We can get project via mentors but Mentor - Project is many-many relation. Is there any other way or table to check if there is any project associated with a course?

You're right, thanks for pointing that out. The table should actually be `projects_courses` instead of `projects_mentors`, I have updated it now.

> > the number of hours corresponding to the lessons completed so far (or within the date range specified for the mentors table)
> 
> I tried to check in the code to get this but couldn't find any reference of hours related to completed lessons. Please help in providing direction to get this number. Which tables should be referred here?

**Comment by ugudlado** [comment]
**From Edmond:**
The tables prefixed with user_lessons_ are actually obsolete, we are only using the ones prefixed with users_courses_. In order to get the completed lessons, we are just considering one lesson per week after the course start date and counting until the current date. 

We are only using the information from the users_courses (the start_date_time value) and users_courses_lessons_canceled tables to get the number of lessons completed.

**Comment by ManojBahuguna** [comment]
@edmondpr, could you please clarify how we are storing relations between testimonials and course?
or is students_testimonials independent of course for mwb in general?

**Comment by edmondpr** [comment]
We can indeed keep the students testimonials independent from the courses in our DB.

**Comment by ManojBahuguna** [comment]
Should we show testimonials here in that case? It will show same testimonials multiple times if student is in multiple courses. Maybe the Students page is fine for that? We can add a link to student details page.

**Comment by edmondpr** [comment]
Most students enroll in a single course but it's still ok if the same student's testimonial appears in the details of multiple mentors because the students often acknowledge all their mentors in these testimonials.
